File: python/캐시.py
from collections import deque

def solution(cacheSize, cities):
    answer = 0
    cache = deque()
    if cacheSize == 0:
        return 5 * len(cities)
    
    for city in cities:
        city = city.lower()
        if city in cache:
            answer += 1
            cache.remove(city)
            cache.append(city)
        else:
            if len(cache) < cacheSize:
                cache.append(city)
            else:
                cache.popleft()
                cache.append(city)
            answer += 5
    return answer

File: js/캐시.js
function solution(cacheSize, cities) {
    let answer = 0;
    const CACHE_HIT = 1;
    const CACHE_MISS = 5;
    if (cacheSize === 0) return cities.length * CACHE_MISS;
    const Q = [];
    cities.forEach(city => {
        city = city.toLowerCase();
        const idx = Q.indexOf(city);
        if (idx === -1) {
            answer += CACHE_MISS;
            if (Q.length >= cacheSize) Q.shift();
        } else {
            answer += CACHE_HIT;
            Q.splice(idx, 1);
        }
        Q.push(city);
    })

    return answer;
}

const cacheSize = 0;
const cities = 	['Jeju', 'Pangyo', 'Seoul', 'NewYork', 'LA']
const answer = solution(cacheSize, cities)
console.log(answer);